lightningd/plugin_hook.c

```diff
@@ -102,13 +121,22 @@ void plugin_hook_call_(struct lightningd *ld, const struct plugin_hook *hook,
 		ph_req = notleak(tal(hook->plugin, struct plugin_hook_request));
 		req = jsonrpc_request_start(NULL, hook->name,
 					    plugin_get_log(hook->plugin),
-					    plugin_hook_callback, ph_req);
+					    sync ? plugin_hook_sync_callback : plugin_hook_callback,
+					    ph_req);
 		ph_req->hook = hook;
 		ph_req->cb_arg = cb_arg;
 		ph_req->db = ld->wallet->db;
 		hook->serialize_payload(payload, req->stream);
 		jsonrpc_request_end(req);
 		plugin_request_send(hook->plugin, req);
+		if (sync) {
+			ret = plugin_exclusive_loop(hook->plugin);
+			if (ret != ph_req) {
+				void *ret2 = plugin_exclusive_loop(hook->plugin);
+				assert(ret2 == ph_req);
+				io_break(ret);
```

Comment:
I... definitely do not understand what this code is doing!

Reply:
I took it from the `db_write` hook but the comment made it out to the next commit.